### PR TITLE
STM32: Add displayio to F7/H7

### PR DESCRIPTION
**Pull request by hierophect**
This PR adds DisplayIO support to the F7 and H7 STM32 families. Should enable some fairly impressive screen use given the large RAM sizes and speeds of these chipsets. 

Includes some minor changes to remove DisplayIO's strict dependency on Pulsio. Before, DisplayIO required PulseIO to be implemented as empty stubs but with the CIRCUITPY_PULSIO macro still turned on, which was confusing from a documentation perspective and not especially necessary since it could smoothly default to DigitalIO backlights already. 

Another minor change included is tripling the default filesystem size of the H7. It may be beneficial in the future to add some kind of board-level setting to adjust filesystem size on these chips, since H7 dev boards often forgo SPI flash since the on-board flash is so huge (1MB). For now it's simply large enough to cleanly fit all the DisplayIO .mpy libraries without crashing. 

Tested on Nucleo H743, Nucleo H767 with a SSD1306 with success.

### CODE REVIEW

File: ports/stm/mpconfigport.mk [hunk 1]
```diff
@@ -50,7 +43,4 @@ ifeq ($(MCU_SERIES),F7)
 	CIRCUITPY_FREQUENCYIO = 0
 	CIRCUITPY_I2CSLAVE = 0
 	# shared-module modules that still need prerequisites filled
```
Comment (tannewt) [inline ports/stm/mpconfigport.mk:45]:
```suggestion
```

File: shared-module/displayio/Display.c [hunk 2]
```diff
@@ -111,6 +111,8 @@ void common_hal_displayio_display_construct(displayio_display_obj_t* self,
     // Always set the backlight type in case we're reusing memory.
     self->backlight_inout.base.type = &mp_type_NoneType;
     if (backlight_pin != NULL && common_hal_mcu_pin_is_free(backlight_pin)) {
+        // Avoid PWM types and functions when the module isn't enabled
+        #if (CIRCUITPY_PULSIO)
```
Comment (tannewt) [inline shared-module/displayio/Display.c:115]:
```suggestion
        #if (CIRCUITPY_PULSEIO)
```

Please test on a non-STM board as well with a display that PWMs the backlight. The #ifs here have a typo.

Comment (hierophect) [reply shared-module/displayio/Display.c:115]: Whoops, good point and good catch. I'll make a note to include an out-of-port control when testing shared module.

**Comment by hierophect** [comment]
@tannewt ping for re-review

**Comment by hierophect** [comment]
Tested on STM32 Feather F405 and Atmel-SAMD Feather M4 Express for regression, works on both.

**Pull request merged**